digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/AssetFiller.scala

```diff
@@ -568,3 +568,37 @@ object NumericalLimitFiller {
       informationSource = asset.informationSource), changeSet)
   }
 }
+
+class OneWayAssetFiller extends AssetFiller {
+  override protected def combineEqualValues(segmentPieces: Seq[SegmentPiece], segments: Seq[PersistedLinearAsset]): Seq[SegmentPiece] = {
+    def chooseSegment(seg1 :SegmentPiece, seg2: SegmentPiece): Seq[SegmentPiece] = {
+      val sl1 = segments.find(_.id == seg1.assetId).get
+      val sl2 = segments.find(_.id == seg2.assetId).get
+      if (sl1.startMeasure.equals(sl2.startMeasure) && sl1.endMeasure.equals(sl2.endMeasure)) {
+        val winner = segments.filter(l => l.id == seg1.assetId || l.id == seg2.assetId).sortBy(s =>
+          s.endMeasure - s.startMeasure).head
+        Seq(segmentPieces.head.copy(assetId = winner.id, sideCode = SideCode.BothDirections))
+      } else {
+        segmentPieces
+      }
+    }
+
+    if(segmentPieces.size < 2)
```

Comment:
Since this override is practically equal of the overrided method I suggest something like this:

```suggestion
   class OneWayAssetFiller extends AssetFiller {
        override protected def combineEqualValues(segmentPieces: Seq[SegmentPiece], segments: Seq[PersistedLinearAsset]): Seq[SegmentPiece] = {
               if(segmentPieces.size < 2)
                      return segmentPieces
                      
               super.combineEqualValues(segmentPieces, segments)             
        }
   }
```